### ISSUE TITLE
saltenv url-parameter not working in file.managed for salt:// sources since 2015.8

### ISSUE THREAD

**Issue by HerrBerg** [saltstack/salt#28822]
Until 2015.5 this feature was working fine:

```
salt-call state.single file.managed /tmp/foo source='salt://templates/startscript.jinja?saltenv=meta'
```

Since 2018.8 the source file is not found anymore.
When I use 'env' instead of 'saltenv' a DeprecationWarning is returned that I should use 'saltenv' instead, but the file is not found either.

**Comment by jfindlay** [comment]
@HerrBerg, thanks for the report.  I can reproduce this.
```console
# mkdir /srv/salt/meta
# echo contents > /srv/salt/meta/file
```
```console
# salt-call state.single file.managed /tmp/file source='salt://file?saltenv=meta'
[INFO    ] Loading fresh modules for state activity
[INFO    ] Running state [/tmp/file] at time 09:10:49.080623
[INFO    ] Executing state file.managed for /tmp/file
[ERROR   ] Source file salt://file?saltenv=meta not found
[INFO    ] Completed state [/tmp/file] at time 09:10:49.093585
local:
----------
          ID: /tmp/file
    Function: file.managed
      Result: False
     Comment: Source file salt://file?saltenv=meta not found
     Started: 09:10:49.080623
    Duration: 12.962 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:  12.962 ms
```
```console
# salt-call state.single file.managed /tmp/file source='salt://meta/file'
[INFO    ] Loading fresh modules for state activity
[INFO    ] Running state [/tmp/file] at time 09:11:04.034860
[INFO    ] Executing state file.managed for /tmp/file
[INFO    ] Fetching file from saltenv 'base', ** done ** u'meta/file'
[INFO    ] File changed:
New file
[INFO    ] Completed state [/tmp/file] at time 09:11:04.058087
local:
----------
          ID: /tmp/file
    Function: file.managed
      Result: True
     Comment: File /tmp/file updated
     Started: 09:11:04.034860
    Duration: 23.227 ms
     Changes:
              ----------
              diff:
                  New file
              mode:
                  0644

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  23.227 ms
```
```yaml
# salt-call --versions
Salt Version:
           Salt: 2015.8.1

Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
           Mako: 1.0.0
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: debian 8.2
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.2
```

**Comment by robgott** [comment]
I am experiencing the same issue and can trace it back to d70914e47349ba04ec0e5baa59642618cd616cd7 where the `file.get_managed` function changed so that it would start calling `cp.hash_file` which does not implement any `salt.util.url.split_env` type call to look at the saltenv query parameter.
@jfindlay - I can see that you have added this functionality to the `cp.get_file` function so I've added it to `cp.hash_file` also but this means it would be called twice for a `get_file` call. It would be good to confirm if it needs to be added to all the other cp functions or if it should be moved out to the file module.  #30166

**Comment by jfindlay** [comment]
@robgott, thanks for your research and work on this.  I changed all of the salt url handling in develop before Beryllium was forked and so only modified code that was already there.  If the env needs to be split out of the URL in `cp.hash_file` then it should also be split in `fileclient.RemoteClient.hash_file` (compare `fileclient.RemoteClient.get_file`).  I am not very familiar with the workings of the `cp` module and the fileclient so I can't say anything about where else this may also need to be, but the efficiency effects should be minor.

As your pull request was submitted against develop, I added the backport label.  I have fixed the RemoteClient function in #30177.
